### PR TITLE
Add a metadata to generate a CNAME file

### PR DESCRIPTION
**Pull request by leoruhland**
This fixes the problem with CNAME and remote templates which is mentioned in the docs/configuration. Setting a cname metadata in couscous.yml generates a CNAME file.

If you find a good idea, I can add some documentation and provide some tests too.

### CODE REVIEW

**Comment by mnapoli** [comment]
Hi, thank you for the pull request, this is an awesome feature, I'd love to merge it! Tests and doc would be welcome indeed.

**Comment by bobmulder** [comment]
I really like it as well! 

@mnapoli I would suggest to improve the docs at all. For me it's not very clear. Also, a section to document how to extend this plugin would be great. @leoruhland figured it out himself (by adding steps), but if you can explain this, I'm pretty sure more extensions will be added ;)

**Comment by leoruhland** [comment]
I made some changes in the step I created, adding the CNAME file to the list of the files to be written instead of directly writing using the FileSystem Class.
I added a test too.

Hey @mnapoli, what do you think? Do you agree with the changes? I am still learning PHPUnit, the test I created is satisfactory?

My english is not very good, maybe someone can help me to add the documentation? @bobmulder :$

File: src/Module/Template/Model/CnameFile.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Couscous\Module\Template\Model;
+
+use Couscous\Model\File;
+
+/**
+ * @author Matthieu Napoli <matthieu@mnapoli.fr>
```
Comment (bobmulder) [inline src/Module/Template/Model/CnameFile.php]: @leoruhland I guess that you created this file, instead of @mnapoli? ;)

**Comment by bobmulder** [comment]
@leoruhland Thanks for noticing. I will try to add some docs tonight ;) I guess it isn't that much, only some docs to inform about the CNAME option and reference to the GH-Pages docs...

**Comment by mnapoli** [comment]
Excellent @leoruhland thank you! @bobmulder I would gladly merge documentation for that!

FYI I'll be merging this pull request probably in a few days (holidays :wink:)

**Comment by bobmulder** [comment]
Nice, I suggest I will add the docs in a seperate PR

**Comment by leoruhland** [comment]
I will add the docs, clean the commits and make a new pull request!

**Comment by bobmulder** [comment]
So docs aren't needed anymore?

**Comment by mnapoli** [comment]
The pull request was reopened in #138, as @leoruhland said he added the docs in there.

**Comment by bobmulder** [comment]
@mnapoli Just a message to warn you, couscous.io is down! The CNAME file isn't located ;)

**Comment by mnapoli** [comment]
@bobmulder thanks! I deployed with an older version so the CNAME wasn't generated haha ;)

**Comment by bobmulder** [comment]
Haha okay. As i mentioned at gitter I used a fork to add the cname so couscous.io is currently used by my fork. Let me know when I should remove it!

**Pull request closed without merging**